pkg/auth/oauth/registry/tokentimeouts.go

```diff
@@ -201,15 +203,27 @@ func (a *oauthTokenTimeoutValidator) flush(flushHorizon time.Time) {
 		patch := []byte(fmt.Sprintf(timeoutsInPatch, td.token.TimeoutsIn, newTimeout))
 		_, err := a.tokens.Patch(td.token.Name, ktypes.JSONPatchType, patch)
 		if err != nil {
+			// if the token has been deleted, we should not retry
+			if !kerrors.IsNotFound(err) {
+				failedPatches = append(failedPatches, td)
+			}
 			glog.V(5).Infof("Token timeout for user=%q client=%q scopes=%v was not updated: %v",
 				td.token.UserName, td.token.ClientName, td.token.Scopes, err)
-			// TODO can we not remove and try later?
 		} else {
+			flushedTokens++
 			glog.V(5).Infof("Updated token timeout for user=%q client=%q scopes=%v creation=%s from %d to %d",
 				td.token.UserName, td.token.ClientName, td.token.Scopes, td.token.CreationTimestamp.Time, td.token.TimeoutsIn, newTimeout)
 		}
+
 		a.remove(td, tdr)
-		flushedTokens++
+	}
+
+	if retry := len(failedPatches); retry > 0 {
+		glog.V(5).Infof("Re-adding %d failed patches", retry)
+	}
+	// add the failed attempts back so we can try them the next time we flush
```

Comment:
the next time the token may already be expired, if we have a case were it make sense to retry we need to retry right away, or it is useless to add in the queue

Reply:
We can use the exponential backoff helpers to make this retry right away if you prefer that.  But giving up after one failure does not seem correct (though it is hard to test how often it would happen in a real cluster).